Update TripalDocker docs and add in VScode Dev Container information. (#92)
tripal/tripal_doc: docs/contributing/vscode-devcontainer.rst

Using VSCode when developing for Tripal
==========================================

> In addition to using docker directly when testing or developing for Tripal, we also have built in integration with VSCode to make your life that little bit easier! More specifically, we integrate with `VSCode DevContainers <https://code.visualstudio.com/docs/devcontainers/containers>`_ via the devcontainer.json file in the root of this repository. This allows VSCode to automatically:

Reviewer: ```suggestion
In addition to using docker directly when testing or developing for Tripal, we also have built in integration with the `Visual Studio Code <https://code.visualstudio.com/>`_ development environment (VSCode) to make your life that little bit easier! More specifically, we integrate with `VSCode DevContainers <https://code.visualstudio.com/docs/devcontainers/containers>`_ via the devcontainer.json file in the root of this repository. This allows VSCode to automatically:
```

1. Build a docker image based on our Dockerfile
2. Start a container using that image
3. Mount your current local directory inside the container
4. Configure VScode to seamlessly integrate with the container so that Devcontainer enabled extensions use the container directly. For example, opening the terminal will open a bash session directly in the container in your mounted tripal directory.

> 5. Install and configure a number really helpful VSCode extensions to make your life easier including php intelephense (php syntax checking and autocomplete), Drupal specific autocomplete, code standards checking, phpunit testing integration, etc. (see more below)

Reviewer: ```suggestion
In addition to using docker directly when testing or developing for Tripal, we also have built in integration with VSCode to make your life that little bit easier! More specifically, we integrate with `VSCode DevContainers <https://code.visualstudio.com/docs/devcontainers/containers>`_ via the devcontainer.json file in the root of this repository. This allows VSCode to automatically:

1. Build a docker image based on our Dockerfile
2. Start a container using that image
3. Mount your current local directory inside the container
4. Configure VScode to seamlessly integrate with the container so that Devcontainer enabled extensions use the container directly. For example, opening the terminal will open a bash session directly in the container in your mounted tripal directory.
5. Install and configure a number of really helpful VSCode extensions to make your life easier including php intelephense (php syntax checking and autocomplete), Drupal specific autocomplete, code standards checking, phpunit testing integration, etc. (see more below)
```

  When it's complete, you will see the DevContainer status in the bottom left corner which indicates you are using "Dev Container @ TripalDocker".

3. If you don't see the popup above, you can instead click on the Dev Container status icon in the bottom left corner. This will open the command palette where you can choose "Reopen in Container".

  .. image:: vscode.devcontainer.statusIcon.png

  .. image:: vscode.devcontainer.commandpalette.png

Tour of Devcontainer Features/Benefits
---------------------------------------

Terminal mounts inside container
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

In VSCode, go to Terminal in the top menu bar and click "New Terminal". This will open a terminal inside the container which allows you to use Drush natively without the need for "docker exec" or any other command prefixes!

.. image:: vscode.devcontainer.terminal.drush.png

Ports automatically mapped
^^^^^^^^^^^^^^^^^^^^^^^^^^^

Ports exposed via the dockerfile are now mapped automatically to ports on your local machine. This allows you to have multiple devcontainers open without having to think about ports at all.

> To view your Drupal site, go to "Ports", hoveer over the forwarded address for port 80 which runs the webserver and then click on the little globe icon. This will open the Drupal homepage in your default web browser. Note the login information was printed to the terminal during devcontainer startup *winks*.

Reviewer: ```suggestion
To view your Drupal site, go to "Ports", hover over the forwarded address for port 80 which runs the webserver, and then click on the little globe icon. This will open the Drupal homepage in your default web browser. Note the login information was printed to the terminal during devcontainer startup 😉.
```
Did you mean the winks emoji here?

Author: Yup but I didn't know how to add it in and didn't want to take the time to look it up.

While it does have a premium version, we use the free version which includes the following:

 - Fast camel/underscore case code completion (IntelliSense) for document, workspace and built-in symbols and keywords with automatic addition of use declarations.
 - Detailed signature (parameter) help for document, workspace and built-in constructors, methods, and functions.
 - Rapid workspace wide go to definition support.
 - Workspace wide find all references.
 - Fast camel/underscore case workspace symbol search.
 - Full document symbol search that also powers breadcrumbs and outline UI.
 - Multiple diagnostics for open files via an error tolerant parser and powerful static analysis engine.
 - Lossless PSR-12 compatible document/range formatting. Formats combined HTML/PHP/JS/CSS files too.
 - Embedded HTML/JS/CSS code intelligence.
 - Detailed hover with links to official PHP documentation.
 - Smart highlight of references and keywords.
 - Advanced PHPDoc type system supporting templates and callable signatures.
 - Reads PHPStorm metadata for improved type analysis and suggestions.

**Hover over a method name to get the full documentation header for it.**

> In the screenshot below we simply hovered over the "getFormId()" method and the full documentation from the Drupal API automatically popped up describing it's function, parameters and return value.

Reviewer: ```suggestion
In the screenshot below we simply hovered over the "getFormId()" method, and the full documentation from the Drupal API automatically popped up describing its function, parameters and return value.
```
